frontend/app/components/auth/components/oauth.tsx

```diff
@@ -39,24 +43,47 @@ export function OAuth({ providers }: Props) {
     dispath(setUser(user));
   };
 
+  const handleTelegramClick: JSX.EventHandler<JSX.TargetedMouseEvent<HTMLButtonElement>> = async (evt) => {
+    evt.preventDefault();
+    if (!telegramParams) {
+      const params = await getTelegramSigninParams();
+      if (params === null) {
+        return;
+      }
+      dispath(setTelegramParams(params));
+    }
+    toggleTelegram && toggleTelegram(true);
+  };
+
   return (
     <ul className={clsx('oauth', styles.root)}>
       {providers.map((p) => {
         const { name, icon } = getProviderData(p, theme);
 
         return (
           <li key={name} className={clsx('oauth-item', styles.item)}>
-            <a
-              target="_blank"
-              rel="noopener noreferrer"
-              href={`${BASE_URL}/auth/${p}/login?from=${location}&site=${siteId}`}
-              onClick={handleOathClick}
-              className={clsx('oauth-button', styles.button, styles[buttonVariant], styles[p])}
-              data-provider-name={name}
-              title={intl.formatMessage(messages.oauthTitle, { provider: name })}
-            >
-              <img className="oauth-icon" src={icon} width="20" height="20" alt="" aria-hidden={true} />
-            </a>
+            {name === 'Telegram' ? (
```

Comment:
Maybe it would be better to treat telegram as any other provider and open a link to the bot immediately.
At the same time, we can change the view in the dropdown for telegram confirmation.